### DISCUSSION TITLE
Introducing a suffix for 'byte' literals

### DISCUSSION THREAD

**Discussion by light-traveller** [dotnet/csharplang#6774] [Language Ideas]
### Proposal
Adding a suffix to C# language to explicitly denote the type of a `byte` literal.

Proposed letter for the suffix: **y** and **Y**

The character `'y'` as the second letter in the `byte` keyword is a good choice and does not interfere with the hexadecimal `b` literal.

**Example**
```
var myByte = 0xF5y;
var yourByte = 0b_1010_1100Y;
var anotherByte = 100Y;
```

### Motivation
This will be much of a big help when declaring implicitly typed variables in *bit/byte manipulating scenarios*—in which hard-coded literals are frequently used—by reducing explicit casting and saving key strokes.

**Example**
```c#
var highByte = 0xF0y;
var lowByte = 0x03y;

var bytes = new[] { highByte, lowByte };
var result = Process(bytes);

byte[] Process(byte[] input)
{
    //...
}
```
In the above example, without the proposed suffix, `masks` array will be an `int[]` and the code cannot be compiled.

**Comment by yaakov-h** [comment]
How does:

```csharp
var highByte = 0xF0y;
```

help with "saving key strokes" over the more traditional:

```csharp
byte highByte = 0xF0;
```

?

**Reply by FaustVX** [reply]
I don't think saving key-strokes is a big deal, but consistency is:
```C#
var a = 5Ul;
var b = 5y;
// instead of
byte b = 5;
```
also for method overloading:
```C#
void Func(int a) { }
void Func(byte a) { }

Func(5y);
//instead of
Func((byte)5);
```

**Reply by light-traveller** [reply]
@yaakov-h, @FaustVX 

1. Using `var` is preferred and more modern and easier to maintain. Besides, explicitly typed declarations are out of scope of this argument, because in that case, suffixes for other types such as long (`long dist = 1250;` vs. `var dist = 1250L;`) would be useless. The same reasoning applies to `byte`.
2. Typing each parenthesis necessitates holding the *shift* key.
3. Explicit casting is ugly and adding it is tedious.

So, in addition to consistency, there are other benefits.

**Comment by svick** [comment]
This is a duplicate of https://github.com/dotnet/csharplang/discussions/6496.
